Kernel/FileSystem/Ext2FS/FileSystem.cpp

```diff
@@ -589,14 +589,14 @@ ErrorOr<void> Ext2FS::prepare_to_clear_last_mount(Inode& mount_guest_inode)
     if (any_inode_busy)
         return EBUSY;
 
-    BlockBasedFileSystem::remove_disk_cache_before_last_unmount();
     m_inode_cache.clear();
     m_root_inode = nullptr;
 
     // Mark filesystem as valid before unmount.
     dmesgln("Ext2FS: Clean unmount, setting superblock to valid state");
     m_super_block.s_state = EXT2_VALID_FS;
     TRY(flush_super_block());
+    BlockBasedFileSystem::remove_disk_cache_before_last_unmount();
```

Comment:
The commit message indicates that the `flush_super_block` call goes through the disk cache. How do we ensure that this cache is flushed and the AsyncDeviceRequests have completed before the system is shut down?

Reply:
As far as I can tell the call to flush the superblock will create an async IO request and will block until it is completed.